modules/core/src/v2/coins/eos.ts

```diff
@@ -284,4 +325,81 @@ export class Eos extends BaseCoin {
     };
     return { halfSigned: txParams };
   }
+
+  /**
+   * Deserialize a transaction
+   * @param txHex
+   * @param transactionHeaders
+   */
+  private deserializeTransaction({ txHex, transactionHeaders }: ExplainTransactionOptions): Bluebird<DeserializedEosTransaction> {
+    return co(function *() {
+      const eosClientConfig = { chainId: this.getChainId(), transactionHeaders };
+      const eosClient = new EosJs(eosClientConfig);
+
+      // Get tx base values
+      const eosTxStruct = eosClient.fc.structs.transaction;
+      const serializedTxBuffer = Buffer.from(txHex, 'hex');
+      const transaction = EosJs.modules.Fcbuffer.fromBuffer(eosTxStruct, serializedTxBuffer);
+
+      // Get transfer action values
+      // Only support transactions with one action: transfer
+      if (transaction.actions.length !== 1) {
+        throw new Error(`invalid number of actions: ${transaction.actions.length}`);
+      }
+      const txAction = transaction.actions[0];
+      if (!txAction) {
+        throw new Error('missing transaction action');
+      }
+      if (txAction.name !== 'transfer') {
+        throw new Error(`invalid action: ${txAction.name}`);
+      }
+      const transferStruct = eosClient.fc.abiCache.abi('eosio.token').structs.transfer;
+      const serializedTransferDataBuffer = Buffer.from(txAction.data, 'hex');
+      const transferActionData = EosJs.modules.Fcbuffer.fromBuffer(transferStruct, serializedTransferDataBuffer);
+      transaction.address = transferActionData.to;
+      transaction.amount = this.bigUnitsToBaseUnits(transferActionData.quantity.split(' ')[0]);
+
+      // Get the tx id if tx headers were provided
+      if (transactionHeaders) {
+        const rebuiltTransaction = yield eosClient.transaction(
+          { actions: transaction.actions },
+          { sign: false, broadcast: false },
+        );
+        transaction.transaction_id = rebuiltTransaction.transaction_id;
+      }
+
+      return transaction;
+    }).call(this);
+  }
+
+  /**
+   * Explain/parse transaction
+   * @param params - ExplainTransactionOptions
+   * @returns Object containing transaction explanation
+   */
+  explainTransaction(params: ExplainTransactionOptions): Bluebird<TransactionExplanation> {
+    return co(function *() {
+      let transaction;
+      try {
+        transaction = yield this.deserializeTransaction(params);
+      } catch (e) {
+        // throw new Error('txHex needs to be a hex string');
```

Comment:
Please clean up leftover comment, and if you really just want to rethrow, then the whole try catch is unnecessary. Maybe consider throwing a custom error type instead?